lib/artifactory/resources/build.rb

```diff
@@ -95,24 +95,24 @@ def from_hash(hash, options = {})
     end
 
     # Based on https://github.com/JFrogDev/build-info/blob/master/README.md#build-info-json-format
-    attribute :properties, {}
+    attribute :properties, {}.freeze
```

Comment:
Why `freeze` these instead of creating a `Proc` that would return an empty `Hash`?

Because the `Proc` approach could silently drop data.  For example, if a user did this:

```ruby
build = Artifactory::Resource::Build.new
build.properties['key'] = 'value'
puts build.properties
```

It would print `{}` instead of `{"key" => "value}` like a user would expect.  This happens because the user modified the mutable default returned by the getter, not an object stored in the object's attribute hash.

If this method of accessing and modifying attributes is common, then the getter could be enhanced to store that default value into the object's attribute hash before returning it.